R/index_nhdplus.R

```diff
@@ -77,63 +81,67 @@ matcher <- function(coords, points, search_radius, max_matches = 1) {
 #'
 #' source(system.file("extdata", "sample_flines.R", package = "nhdplusTools"))
 #'
-#' get_flowline_index(sample_flines,
-#'                    sf::st_sfc(sf::st_point(c(-76.87479,
-#'                                              39.48233)),
-#'                               crs = 4326))
+#' point <- sf::st_sfc(sf::st_point(c(-76.87479, 39.48233)),
+#'                     crs = 4326)
 #'
-#' get_flowline_index("download_nhdplusv2",
-#'                    sf::st_sfc(sf::st_point(c(-76.87479,
-#'                                              39.48233)),
-#'                               crs = 4326))
+#' get_flowline_index(sample_flines, point)
 #'
-#' get_flowline_index(sample_flines,
-#'                    sf::st_sfc(sf::st_point(c(-76.87479,
-#'                                              39.48233)),
-#'                               crs = 4326), precision = 30)
+#' point <- sf::st_transform(point, 5070)
+#'
+#' get_flowline_index(sample_flines, point,
+#'                    search_radius = units::set_units(200, "m"))
+#'
+#' get_flowline_index("download_nhdplusv2", point)
+#'
+#' get_flowline_index(sample_flines, point, precision = 30)
 #'
 #' get_flowline_index(sample_flines,
 #'                    sf::st_sfc(list(sf::st_point(c(-76.86934, 39.49328)),
 #'                                    sf::st_point(c(-76.91711, 39.40884)),
 #'                                    sf::st_point(c(-76.88081, 39.36354))),
 #'                               crs = 4326),
-#'                    search_radius = 0.2,
+#'                    search_radius = units::set_units(0.2, "degrees"),
 #'                    max_matches = 10)
 #'
 #' }
 
 get_flowline_index <- function(flines, points,
-                               search_radius = 0.1,
+                               search_radius = NULL,
                                precision = NA,
                                max_matches = 1) {
 
+  in_crs <- sf::st_crs(points)
+
+  search_radius <- check_search_radius(search_radius, points)
```

Comment:
This is the main thing that got added -- see the function declaration down below this function.